### PR TITLE
[intel-npu] Implementing DEVICE_GOPS metric

### PR DESCRIPTION
**Pull request by csoka**
### Details:
 - Implemeting DEVICE_GOPS metric for intel-npu plugin

### Tickets:
 - EISW-123380

### CODE REVIEW

File: src/plugins/intel_npu/src/backend/src/zero_device.cpp [hunk 1]
```diff
@@ -46,6 +46,14 @@ ZeroDevice::ZeroDevice(const std::shared_ptr<ZeroInitStructsHolder>& initStructs
                        ze_result_to_description(retpci));
     }
 
+    /// Calculate and store device GOPS with formula: frequency * number of tiles * ops per tile
+    float gops = (device_properties.coreClockRate / powf(1000, 3)) * device_properties.numSlices *
+                 device_properties.physicalEUSimdWidth;
+    device_gops[ov::element::f32] = 0;
+    device_gops[ov::element::u8] = gops;
+    device_gops[ov::element::i8] = gops;
+    device_gops[ov::element::f16] = 0.5f * gops;
```
Comment (ArtemySkrebkov) [inline src/plugins/intel_npu/src/backend/src/zero_device.cpp:55]: for my education, why do we multiple by 0.5 for fp16?

**Review by ArtemySkrebkov — APPROVED** [review]

File: src/plugins/intel_npu/src/backend/src/zero_device.cpp (hunk 1 above)
Comment (csoka) [reply src/plugins/intel_npu/src/backend/src/zero_device.cpp:55]: this is based on the indications from the Arch team

**Review by razvanapetroaie — APPROVED** [review]
Ticket in description if you don't mind

**Comment by csoka** [comment]
I still need to add a cross-OS backwards compatibility check, we are currently working on getting alined-with with the driver teams, as older drivers don't report perf/slice in physicalEUSimdWidth, leading to incorrect numbers.

**Pull request merged**